libcontainerd/types/types_linux.go

```diff
@@ -12,14 +11,17 @@ type Summary struct{}
 
 // Stats holds metrics properties as returned by containerd
 type Stats struct {
-	Read    time.Time
-	Metrics *statsV1.Metrics
+	Read time.Time
+	// Metrics is expected to be either one of:
+	// * github.com/containerd/cgroups/stats/v1.Metrics
+	// * github.com/containerd/cgroups/stats/v2.Metrics
+	Metrics interface{}
```

Comment:
Could this be 2 separate fields instead?

Reply:
I think `interface{}` is better for extensibility. Potentially non-runc runtime may produce other type of Metrics.

Reply:
How would we consume that?

Reply:
We would need  to add `switch Metrics.(type)` cases as in this PR.